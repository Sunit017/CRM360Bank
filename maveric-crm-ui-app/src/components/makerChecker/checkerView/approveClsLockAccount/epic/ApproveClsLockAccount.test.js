import ApproveClsLockAccount from "..";
import renderWithRedux, { renderWithProviders } from "../../../../../testUtil";
import { cleanup, fireEvent, screen } from "@testing-library/react";
import React from "react";
import userEvent from "@testing-library/user-event";

afterEach(cleanup);

test("Render Approve/Close/Lock/Unlock screen", () => {
  const container = renderWithProviders(
    <ApproveClsLockAccount
      checkerHeight="200px"
      isChecker={false}
      checkerData={null}
    />,
    {}
  );
  expect(container).toMatchSnapshot();
});

test("Make  Approve/Close/Lock/Unlock form fields  ", () => {
  const { container } = renderWithProviders(
    <ApproveClsLockAccount
      checkerHeight="200px"
      isChecker={false}
      checkerData={null}
    />,
    {
      preloadedState: {
        customerSelection: { data: "1234" },
        account360Search: { accountID: "test_accountID_1" },
        account360ClientDetails: { id: "test_client_id_1" },
      },
    }
  );
  const operationType = container.querySelector(
    'input[name="operationType"]'
  );
  fireEvent.change(operationType, {
    target: {
      value: "Lock",
    },
  });

  const submit = container.querySelector('button[type="submit"]');
  fireEvent.click(submit);

  const reset = container.querySelector('button[type="reset"]');
  fireEvent.click(reset);
});


test("Checker Approve Button  Approve/Close/Lock/Unlock ", async () => {
  const { container } = renderWithProviders(
    <ApproveClsLockAccount
      checkerHeight="200px"
      isChecker={true}
      checkerData={{ customerId: "test_customerID_1", accountNo: "test_accountNo_1", caseId: "test_caseID_1", accountState: "ACTIVE" }}
    />,
    {}
  );
  
  const approveButton = await screen.findByText("Approve");
  userEvent.click(approveButton);
});

test("Checker Reject Button Approve/Close/Lock/Unlock", async () => {
  const { container } = renderWithRedux(
    <ApproveClsLockAccount
      checkerHeight="200px"
      isChecker={true}
      checkerData={{}}
    />,
    {}
  );

  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
  const onReject = jest.fn();

  const rejectButton = await screen.findByText("Reject");
  userEvent.click(rejectButton);
});
