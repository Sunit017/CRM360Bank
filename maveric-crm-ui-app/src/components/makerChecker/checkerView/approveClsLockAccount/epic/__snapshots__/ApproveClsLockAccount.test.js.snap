// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Approve/Close/Lock/Unlock screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: fixed; z-index: 9999; top: 85px; left: 20px; right: 16px; bottom: 16px; pointer-events: none;"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-3t3wfi-MuiGrid-root"
        >
          <div
            class="sc-bczRLJ ixhord"
            style="font-size: 14px;"
          >
            Approve / Close / Lock / Unlock Account
             
             - Maker
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-h9pj2j-MuiGrid-root"
        >
          <div
            class="sc-bczRLJ ixhord"
            style="background: rgb(255, 194, 60);"
          >
            Account Number: 
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-h9pj2j-MuiGrid-root"
        >
          <div
            class="sc-bczRLJ ixhord"
            style="background: rgb(255, 194, 60);"
          >
            Account Holder Name: 
            undefined undefined undefined
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-h9pj2j-MuiGrid-root"
        >
          <div
            class="sc-bczRLJ ixhord"
            style="background: rgb(255, 194, 60);"
          >
            Account State: 
          </div>
        </div>
      </div>
      <div
        style="margin-top: 8px; padding: 16px 8px; border: 1px solid black; height: calc(100vh - 200px); overflow: auto;"
      >
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
            >
              <div
                class="sc-dkzDqf cGxUWW"
              >
                <label
                  class="required"
                >
                  Operation Type
                </label>
                <div
                  class="MuiFormControl-root MuiTextField-root sc-hKMtZM evXmLm css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="operationType"
                      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      id="operationType"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      name="operationType"
                      tabindex="-1"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-hdw1oc"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="sc-gsnTZi diBaTH"
              >
                Notes
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="sc-dkzDqf cGxUWW"
                style="grid-template-columns: 0.163fr 1fr; column-gap: 4px;"
              >
                <label>
                  Notes/Narrative
                </label>
                <div
                  class="MuiFormControl-root MuiTextField-root sc-hKMtZM evXmLm css-xwwqvg-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                      id="notes"
                      name="notes"
                      rows="2"
                      style="height: 0px; overflow: hidden;"
                    />
                    <textarea
                      aria-hidden="true"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                      readonly=""
                      style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                      tabindex="-1"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-hdw1oc"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="display: flex; align-items: center; justify-content: space-evenly; margin-top: 16px; padding: 16px 8px 1px 8px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Submit
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="reset"
            >
              Reset
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; z-index: 9999; top: 85px; left: 20px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-3t3wfi-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ ixhord"
          style="font-size: 14px;"
        >
          Approve / Close / Lock / Unlock Account
           
           - Maker
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-h9pj2j-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ ixhord"
          style="background: rgb(255, 194, 60);"
        >
          Account Number: 
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-h9pj2j-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ ixhord"
          style="background: rgb(255, 194, 60);"
        >
          Account Holder Name: 
          undefined undefined undefined
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-h9pj2j-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ ixhord"
          style="background: rgb(255, 194, 60);"
        >
          Account State: 
        </div>
      </div>
    </div>
    <div
      style="margin-top: 8px; padding: 16px 8px; border: 1px solid black; height: calc(100vh - 200px); overflow: auto;"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
          >
            <div
              class="sc-dkzDqf cGxUWW"
            >
              <label
                class="required"
              >
                Operation Type
              </label>
              <div
                class="MuiFormControl-root MuiTextField-root sc-hKMtZM evXmLm css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="operationType"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="operationType"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    name="operationType"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="sc-gsnTZi diBaTH"
            >
              Notes
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="sc-dkzDqf cGxUWW"
              style="grid-template-columns: 0.163fr 1fr; column-gap: 4px;"
            >
              <label>
                Notes/Narrative
              </label>
              <div
                class="MuiFormControl-root MuiTextField-root sc-hKMtZM evXmLm css-xwwqvg-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    id="notes"
                    name="notes"
                    rows="2"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center; justify-content: space-evenly; margin-top: 16px; padding: 16px 8px 1px 8px;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Submit
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="reset"
          >
            Reset
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
